src/recipes/engine-singleton.md

```diff
@@ -81,16 +82,52 @@ unsafe impl ExtensionLibrary for MyExtension {
 
     fn on_level_deinit(level: InitLevel) {
         if level == InitLevel::Scene {
-            // Unregistering is needed to avoid memory leaks and 
-            // warnings, especially for hot reloading.
-            Engine::singleton().unregister_singleton(
-                StringName::from("MyEditorSingleton")
-            );
+            // Retrieve and keep the pointer to the `Engine` singleton, as
+            // we will need it for some operations.
+            let mut engine = Engine::singleton();
+
+            // To avoid unnecessary construction of the StringName multiple times.
+            let singleton_name = StringName::from("MyEditorSingleton");
+
+            // We need to retrieve the pointer to the singleton object,
+            // as it has to be freed manually - unregistering singleton 
+            // doesn't do it automatically.
+            let singleton = engine
+                .get_singleton(singleton_name.clone())
+                .expect("cannot retrieve the singleton");
+
+            // Unregistering singleton and freeing the object itself is needed 
+            // to avoid memory leaks and warnings, especially for hot reloading.
+            engine.unregister_singleton(singleton_name.clone());
+            singleton.free();
         }
     }
 }
 ```
 
+```admonish warning
+As seen from above, the Godot engine, after singleton registration, isn't managing the singleton pointer. It has some implications for objects 
+whose class inherits from `RefCounted`, namely: Godot won't keep the object alive, which can lead to errors on its retrievals.
+
+Until `gdext` presents a more ergonomic API for singletons, the current solution used in the community is to wrap the pointer in a facade class 
+that itself doesn't inherit from `RefCounted`. Afterwards, the wrapper itself should be registered as a singleton.
+```
+
+```rust
+#[derive(GodotClass)]
+#[class(tool, init, base=RefCounted)]
+struct MyEditorSingleton {
+    base: Base<RefCounted>,
+}
+
+#[derive(GodotClass)]
+#[class(tool, init, base=Object)]
+struct MyEditorSingletonWrapper {
+    // Annotation allows exposing the RefCounted object on the GDScript side.
+    #[var]
+    inner: Gd<MyEditorSingleton>
+}
+```
```

Comment:
`tool` was an omission - I've copied the declaration from the code block above. I guess both me and the initial author of this chapter were using singletons for some kind of editor tools, which was the source of the mistake. Will remove it also from the mentioned code block, as it isn't needed. I think I will include information that `tool` needs to be added if the singleton is intended for usage in-editor.

For this code block: I've intended this as a way to show the workaround if you **need** to make an object of a specific class a singleton, and this class happens to be inheriting from `RefCounted` (which happened to me during `gd-props` development). On second thought it may be a little too verbose for this chapter given that the circumstances are pretty specific. Hyperlink to the `gdext` issue you've mentioned above should be enough and will keep all of the information inside the warning block.